gadopt/diagnostics.py

```diff
@@ -102,3 +118,14 @@ def ux_max(self, boundary_id=None) -> float:
             ux_data = self.u.dat.data_ro[:, 0]
 
         return self.u.comm.allreduce(ux_data.max(), MPI.MAX)
+
+    def uv_min(self, boundary_id=None) -> float:
+        "Minimum value of vertical component of velocity/displacement"
+        self.uv.interpolate(vertical_component(self.u))
+        if boundary_id:
+            bcu = DirichletBC(self.uv.function_space(), 0, boundary_id)
+            uv_data = self.uv.dat.data_ro_with_halos[bcu.nodes]
```

Comment:
This could run into the same issue as #319.

Reply:
doesn't adding 

```
uv_data.min(initial=np.inf)
```

fix this? I think this was the fix suggedted in #299 ...